beetsplug/parentwork.py

```diff
@@ -0,0 +1,197 @@
+                    parent_composer.append(artist['artist']['name'])
+                    parent_composer_sort.append(artist['artist']['sort-name'])
+        if not composer_exists:
+            self._log.info(item.artist + ' - ' + item.title)
+            self._log.debug(
+                "no composer, add one at https://musicbrainz.org/work/" +
+                work_info['work']['id'])
+        parentwork = work_info['work']['title']
+        mb_parentworkid = work_info['work']['id']
+        if 'disambiguation' in work_info['work']:
+            parentwork_disambig = work_info['work']['disambiguation']
+            return [parentwork, mb_parentworkid, parent_composer,
+                    parent_composer_sort, parentwork_disambig]
+        else:
+            return [parentwork, mb_parentworkid, parent_composer,
+                    parent_composer_sort, None]
+
+    def find_work(self, item, force):
+
+        recording_id = item.mb_trackid
+        try:
+            item.parentwork
+            hasparent = True
+        except AttributeError:
+            hasparent = False
+        if not item.mb_workid:
+            self._log.info("No work attached, recording id: " +
+                           recording_id)
+            self._log.info(item.artist + ' - ' + item.title)
+            self._log.info("add one at https://musicbrainz.org" +
+                           "/recording/" + recording_id)
+            return
+        found = False
+        if force or (not hasparent):
+            try:
+                work_info, work_date = find_parentwork(item.mb_workid)
+                parent_info = self.get_info(item, work_info)
+                parentwork = parent_info[0]
+                mb_parentworkid = parent_info[1]
+                parent_composer = parent_info[2]
+                parent_composer_sort = parent_info[3]
+                parentwork_disambig = parent_info[4]
+
+                found = True
+            except musicbrainzngs.musicbrainz.WebServiceError:
+                self._log.debug("Work unreachable")
+                found = False
+        elif hasparent:
+            self._log.debug("Work already in library, not necessary fetching")
+            return
+
+        if found:
+            self._log.debug("Finished searching work for: " +
+                            item.artist + ' - ' + item.title)
+            self._log.debug("Work fetched: " + parentwork +
+                            ' - ' + u', '.join(parent_composer))
+            item['parentwork'] = parentwork
+            if parentwork_disambig:
+                item['parentwork_disambig'] = parentwork_disambig
+            item['mb_parentworkid'] = mb_parentworkid
+            item['parent_composer'] = u''
+            item['parent_composer'] = u', '.join(parent_composer)
+            item['parent_composer_sort'] = u''
+            item['parent_composer_sort'] = u', '.join(parent_composer_sort)
+            if work_date:
+                item['work_date'] = work_date
```

Comment:
Maybe there's a simpler way to do all these assignments—something along the lines of:

```py
for key, value in info.items():
    if value:
        item[key] = value
```

…especially if the return type of `get_info` is changed to a dict.